From 4ad1d3be11d5c421567134a53384374a81ad3db4 Mon Sep 17 00:00:00 2001
From: Michal Mazur <mkm@semihalf.com>
Date: Wed, 31 Jan 2018 16:32:54 +0100
Subject: [PATCH 1/4] linux-generic: pool: Return address range in pool info

Implement support in odp_pool_info API to provide address range
of application accessible pool data (linux-generic platform).

Pull request of API change for reference:
https://github.com/Linaro/odp/pull/200

Signed-off-by: Michal Mazur <mkm@semihalf.com>
---
 platform/linux-generic/odp_pool.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/platform/linux-generic/odp_pool.c b/platform/linux-generic/odp_pool.c
index e20693d..a33dd45 100644
--- a/platform/linux-generic/odp_pool.c
+++ b/platform/linux-generic/odp_pool.c
@@ -692,6 +692,9 @@ int odp_pool_info(odp_pool_t pool_hdl, odp_pool_info_t *info)
 	if (pool->params.type == ODP_POOL_PACKET)
 		info->pkt.max_num = pool->num;
 
+	info->min_data_addr = (uint64_t) pool->base_addr;
+	info->max_data_addr = (uint64_t) pool->base_addr + pool->shm_size - 1;
+
 	return 0;
 }
 
-- 
2.7.4

